src/Orleans/Streams/PersistentStreams/IStreamQueueBalancer.cs

```diff
@@ -8,27 +8,32 @@ namespace Orleans.Streams
     /// The stream queue balancer is responsible for load balancing queues across all other related queue balancers.  It
     /// notifies any listeners (<code>IStreamQueueBalanceListener</code>) of changes to the distribution of queues.
     /// </summary>
-    internal interface IStreamQueueBalancer
+    public interface IStreamQueueBalancer
     {
         /// <summary>
+        /// Initialize this instance
+        /// </summary>
+        /// <returns></returns>
+        Task Initialize();
+        /// <summary>
         /// Retrieves the latest queue distribution for this balancer.
         /// </summary>
         /// <returns>Queue allocated to this balancer.</returns>
-        IEnumerable<QueueId> GetMyQueues();
+        Task<IEnumerable<QueueId>> GetMyQueues();
```

Comment:
Ah, sorry, I missed that.  Now I understand.
Ok, so we'd considered this in the original pattern, and chose for this to be synchronous.  The reason for this is that this is a push+pull pattern.  We notify the IStreamQueueBalanceListeners once we get a change, so we already have the change at that point, so there is no reason to be asynchronous.  This is actually important, because the agent is reentrant and giving up it's turn to get the queues could have unexpected side effects.  Even the subscribe and unsubscribe calls where expected to be local, because the queue balancer is expected to be per silo, we don't want it to be distributed and asynchronous.